indico/web/flask/templating.py

```diff
@@ -194,6 +196,43 @@ def call_template_hook(*name, **kwargs):
         return Markup(u'\n').join(x[1] for x in values) if values else u''
 
 
+class CustomizationLoader(BaseLoader):
+    def __init__(self, fallback_loader, customization_dir, customization_debug=False):
+        from indico.core.logger import Logger
+        self.logger = Logger.get('customization')
+        self.debug = customization_debug
+        self.fallback_loader = fallback_loader
+        self.fs_loader = FileSystemLoader(customization_dir, followlinks=True)
+
+    def _get_fallback(self, environment, template, path):
+        rv = self.fallback_loader.get_source(environment, template)
+        if self.debug:
+            try:
+                orig_path = rv[1]
+            except TemplateNotFound:
+                orig_path = None
+            self.logger.debug('Customizable: %s (original: %s, reference: ~%s)', path, orig_path, template)
+        return rv
+
+    def get_source(self, environment, template):
+        path = posixpath.join(*split_template_path(template))
+        if template[0] == '~':
```

Comment:
I don't like this very much but being able to inherit from the original template is important - and from within a loader we have no way of knowing from where the template is being rendered.